src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -148,7 +148,7 @@ impl Step for ToolBuild {
             &self.extra_features,
         );
 
-        if path.ends_with("/rustdoc") &&
+        if (path.ends_with("/rustdoc") || tool == "rust-analyzer") &&
```

Comment:
is there any tool that we *dont* want to apply LTO to? thin lto should be fairly cheap, right?

Reply:
After all, why not. Rustc tools could use the extra perf. I'll check the impact on CI time.